### PR TITLE
 Added statistics for time spent in symbol locks. 

### PR DESCRIPTION
**Pull request by odisseus**

### CODE REVIEW

File: src/reflect/scala/reflect/internal/SymbolTable.scala [hunk 1]
```diff
@@ -479,8 +479,22 @@ abstract class SymbolTable extends macros.Universe
     def transform(sym: Symbol, tpe: Type): Type = tpe
   }
   /** The set of all installed infotransformers. */
-  def infoTransformers = Parallel.synchronizeAccess(this){_infoTransformers}
-  def infoTransformers_=(v: InfoTransformer) = Parallel.synchronizeAccess(this){_infoTransformers = v}
+  def infoTransformers = Parallel.synchronizeAccess(this) {
+    val start = if(StatisticsStatics.areSomeColdStatsEnabled) statistics.pushTimer(statistics.symbolOpsStack, statistics.infoTransformersNanos) else null
```
Comment (pkukielka) [inline src/reflect/scala/reflect/internal/SymbolTable.scala]: Please format your code (e.g. space after `if`)

Comment (pkukielka) [inline src/reflect/scala/reflect/internal/SymbolTable.scala]: @romanowski was mentioning to me that some of that blocks are re-entrant. You need some flag to make sure you are not measuring time in sub-sequent blocks multiple times.

Comment (pkukielka) [reply src/reflect/scala/reflect/internal/SymbolTable.scala]: Or is it solved by stackable timer?

**Comment by pkukielka** [comment]
Please provide an example results.
I do not see in the code how/where that times stats are printed out.

**Comment by odisseus** [comment]
```
./run.py -s ~/devel/scalac_perf/build/pack/ -- -Ystatistics

time spent in symbol locks    : 0 spans, ()0.0ms
time spent in Symbol.info     : 8360985 spans, ()11269.432ms aggregate, 7900.875ms specific
time spent in SymbolTable.infoTransformers: 283499 spans, ()11.883ms aggregate, 11.883ms specific

Compilation done in: 61.1133460999 s
```

**Comment by odisseus** [comment]
Reformatted

**Comment by odisseus** [comment]
There should be no problems with nested calls, because timer stacks are used. See `scala/reflect/internal/util/Statistics.scala:256`. I'm not absolutely sure I've used them correctly though.

**Comment by odisseus** [comment]
Apparently the timers add a lot of overhead themselves. I have measured the same compilation with JFR, and `Symbol.info` calls seem to take only about 1% of all time, although this still qualifies it into the top 5 hot methods.

**Pull request closed without merging**